### PR TITLE
Add Back the Contribution Guide in B.io to the Contributing.MD File

### LINKED ISSUE #21004: Adding Back the Contribution Guide in B.io to the Contributing.MD File
**Description:**
We will need to add back the Contribution Guide added to Bio in [1] to the `contributing.md` file in this ballerina-lang repo.

This is because this has repo-specific information and we need to have only generic contributing guidelines in B.io.

[1] https://ballerina.io/contribution-guide/

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by praneesha**
## Purpose
We will need to add back the Contribution Guide added to Bio in [1] to the contributing.md file in this ballerina-lang repo.

This is because this has repo-specific information and we need to have only generic contributing guidelines in B.io.

[1] https://ballerina.io/contribution-guide/

Fixes #21004

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,195 +1,137 @@
-# Contributing to Ballerina
+---
+layout: ballerina-inner-page
+title: Contributing to Ballerina
+permalink: /contribution-guide
+---
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]: Do we need this information for .md file in this repo?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,195 +1,137 @@
...
+      <div class="cBallerina-Box">
+      <a class="cCommunityIcons" href="https://stackoverflow.com/questions/tagged/ballerina"><img src="/img/stackoverflow.svg"></a>
+      </div>
+   </div>
+   <div class="col-sm-12 col-md-3 cBoxContainer cCommunityIconsContainer">
+      <div class="cBallerina-Box">
+      <a class="cCommunityIcons" href="https://groups.google.com/forum/#!forum/ballerina-dev"><img src="/img/google-groups.svg"></a>
+      </div>
+   </div>
+</div>
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]: How the above stuff including images working in the .md file? Seems we need to reformat this to text along with relevant links.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,195 +1,137 @@
...
 
-Building Ballerina requires a Java SE Development Kit (JDK) version 8 to be installed. You can download one from one of the following locations:
+### Set up the Ballerina plugin
 
-* [http://www.oracle.com/technetwork/java/javase/downloads/index.html](http://www.oracle.com/technetwork/java/javase/downloads/index.html)
-* [http://openjdk.java.net/install/index.html](http://openjdk.java.net/install/index.html)
+Currently, Ballerina has the below plugins developed for IntelliJ IDEA and VS Code. 
+- [Ballerina plugin for IntelliJ IDEA](https://plugins.jetbrains.com/plugin/9520-ballerina)
+    
+    For installation instructions, see [Ballerina IntelliJ Plugin Documentation](https://ballerina.io/learn/vscode-plugin).
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]: Incorrect url?

File: CONTRIBUTING.md (hunk 1 above)
Comment (praneesha) [reply CONTRIBUTING.md]: Removed.

File: CONTRIBUTING.md (hunk 3 above)
Comment (praneesha) [reply CONTRIBUTING.md]: Fixed all 4 URLs of the 2 plugin download locations and Docs.

**Comment by anupama-pathirage** [comment]
@praneesha We don't need these icons here. Those were in website repo as it is generating a web page where presentation matters. So please have the details with links without icons

File: CONTRIBUTING.md (hunk 2 above)
Comment (praneesha) [reply CONTRIBUTING.md]: Added links instead of the HTML icons.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,195 +1,116 @@
...
 
-* Navigate to `File | Open` menu.
-* Browse the filesystem and Select Ballerina Project's root directory.
-   * If prompted to import Gradle project select `Auto import`.
-   * Set Project SDK as Java 1.8
+- Join the conversations at:
 
-##### Useful IDEA Plugins
+    - [StackOverflow](https://stackoverflow.com/questions/tagged/ballerina): to get help with Ballerina (use the Ballerina tag for any of your questions.)
+    - [Slack](https://ballerinalang.slack.com/): for real-time discussions with the team and community
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]: Incorrect url? The correct url is `https://ballerina-platform.slack.com/`. Btw, IMO we should have `https://ballerina.io/community/slack/` instead of that because then users can get the invitations by registering.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,195 +1,116 @@
...
+    - [Ballerina-Dev Google Group](https://groups.google.com/forum/#!forum/ballerina-dev): (developer team mailing list) to discuss Ballerina roadmap, features, and related issues
+    - [Twitter](https://twitter.com/ballerinalang): to tweet about Ballerina (use the “#ballerinalang” hashtag).
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]: Seems points are not consistent. Some have ended with a full stop while others are not.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,195 +1,116 @@
...
 
-##### Importing a Ballerina Project
+-  Start with GitHub issues that can be fixed easily:
+    - Browse issues labeled "good first issue" in the <a href="https://github.com/ballerina-platform/ballerina-lang/issues">ballerina-lang</a> repo.
+    - Use comments on the issue itself to indicate that you will be working on it and get guidance and help.
 
-All Ballerina repositories are developed as Maven or Gradle projects. So you can import any Ballerina project to Eclipse similar to any Maven or Gradle project. 
+## Set up the prerequisites
+1. Download [Ballerina](https://ballerina.io) and go through the [learning resources](https://ballerina.io/learn).
+2. Read the <a href="/code-of-conduct">Ballerina Code of Conduct</a>.
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]: Incorrect url?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,195 +1,116 @@
...
 
-* Navigate to **File** → **Import..**
-* Select **Existing Maven Projects** under **Maven**
-* Next, browse the file system, and open the Maven module.
+## Build the source code 
 
-##### Installing ANTLR4 plugin
-If you are working with the grammar, it would be useful to have ANTLR 4 IDE plugin installed for Eclipse. To install this using Eclipse marketplace:
+For instructions, see <a href="/learn/installing-ballerina/#installing-from-source">Installing from source</a>.
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]: Incorrect url?

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -1,195 +1,116 @@
...
-2. Under **ANTLR4** section, select and enable **Parse Tree** view and **Syntax Diagram** views.
+This plugin will be useful to check and validate a [grammar rule](#contributing-to-ballerina-grammar) you wrote. For instructions on installing the ANTLR4 plugin based on your preferred IDE/Editor (which supports antlr 4.x version), go to [ANTLR Documentation](https://www.antlr.org/tools.html).
 
-### Working With the Ballerina Grammar
+## Submit your contribution
 
-Ballerina grammar has been implemented using ANTLR4. It is recommended to have some basic understanding on ANTLR4 grammar syntax and concepts, before working with the Ballerina grammar. “Parr, Terence (January 15, 2013), The Definitive ANTLR 4 Reference” is a good reference to get started with.
+1. Do your changes in the source code.
+    
+    <details border="1">
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]: The boarder is not visiable?

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,195 +1,116 @@
...
+    <summary>If you are contributing to Ballerina Grammar;</summary>
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]: Should this be `:` instead of `;`

File: CONTRIBUTING.md (hunk 4 above)
Comment (praneesha) [reply CONTRIBUTING.md]: Changed to https://ballerina.io/community/slack/.

File: CONTRIBUTING.md (hunk 5 above)
Comment (praneesha) [reply CONTRIBUTING.md]: Removed the fullstops.

File: CONTRIBUTING.md (hunk 6 above)
Comment (praneesha) [reply CONTRIBUTING.md]: Fixed the link: https://ballerina.io/code-of-conduct

File: CONTRIBUTING.md (hunk 7 above)
Comment (praneesha) [reply CONTRIBUTING.md]: Fixed link: https://ballerina.io/learn/installing-ballerina/#installing-from-source

File: CONTRIBUTING.md (hunk 9 above)
Comment (praneesha) [reply CONTRIBUTING.md]: Changed.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -1,195 +1,119 @@
...
-Extract the Ballerina distribution created at `distribution/zip/ballerina/target/ballerina-<version>-SNAPSHOT.zip`. The `zip/ballerina` directory contains the runtime only. `zip/ballerina-tools/` contains the runtime and tools (e.g., Ballerina Composer).
-
-Note: It is possible to face an IOException error stating "Too many open files". This is due to the default number of possible open files being set to a lower number on your operating system than required for Ballerina to be compiled. You may have to increase the number of open files/file descriptors (FD) on your operating system. This is OS dependent and you can search the internet on how to do this for your operating system (Windows/OSX/Linux). You could update the maximum number of open files to 1000000 (or higher).
-
-### Setting up your development environment
-
-#### Setup IntelliJ IDEA
+- [Get started](#get-started)
+- [Set up the Prerequisites](#set-up-the-prerequisites)
+- [Obtain the source code](#obtain-the-source-code)
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]: Link is not working

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -1,195 +1,119 @@
...
+- [Build the project](#build-the-project)
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]: Link is not working.

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -1,195 +1,119 @@
...
+- [Set up the development environment](#set-up-the-development-environment)
+    - [Set up the Ballerina plugin](#set-up-the-ballerina-plugin)
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]: Link is not working

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -1,195 +1,119 @@
...
+    For installation instructions, see [Ballerina IntelliJ Plugin Documentation](https://ballerina.io/learn/intellij-plugin/).
+- [Ballerina plugin for VS Code](https://marketplace.visualstudio.com/items?itemName=ballerina.ballerina)
+    
+    For installation instructions, see [Ballerina VS Code Documentation](https://ballerina.io/learn/vscode-plugin).
 
-To open the plugin views: 
+### Set up the ANTLR4 plugin
 
-1. Navigate to **Window** → **Show View** → **Other**. 
+This plugin will be useful to check and validate a [grammar rule](#contributing-to-ballerina-grammar) you wrote. For instructions on installing the ANTLR4 plugin based on your preferred IDE/Editor (which supports antlr 4.x version), go to [ANTLR Documentation](https://www.antlr.org/tools.html).
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]:
```suggestion
This plugin will be useful to check and validate a [grammar rule](#contributing-to-ballerina-grammar) you wrote. For instructions on installing the ANTLR4 plugin based on your preferred IDE/Editor (which supports ANTLR 4.x version), go to [ANTLR Documentation](https://www.antlr.org/tools.html).
```

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -1,195 +1,119 @@
...
 
-Build process automatically runs all the tests.
+2. Add unit tests accordingly. (The build process will automatically execute all the tests.)
+3. Commit the changes to your fork and push them to the corresponding original repo by sending a Pull Request (PR). 
 
-#### Commit to Your Fork
+    >**Tip:** Follow these commit message requirements:
 
-We follow these commit message requirements:
+    - Separate subject from body with a blank line
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]:
```suggestion
    - Separate subject from the body with a blank line
```

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -1,195 +1,119 @@
...
+    Ballerina grammar has been implemented using ANTLR plugin version 4.5.3. To get a basic understanding of ANTLR grammar syntax and concepts before working with Ballerina grammar, go to [Parr, Terence (January 15, 2013), The Definitive ANTLR 4 Reference](https://www.oreilly.com/library/view/the-definitive-antlr/9781941222621/).
 
-* BallerinaLexer.g4 - Contains the lexer rules for Ballerina grammar. Lexer is responsible for tokenizing an input Ballerina source code.
-* BallerinaParser.g4 - Contains the parser rules. Parser listens to the token stream generated by the lexer. High level grammar productions/abstractions are defined in the parser using those tokens.
+    [Ballerina grammar](https://github.com/ballerina-platform/ballerina-lang/tree/master/compiler/ballerina-lang/src/main/resources/grammar) consists of two files:
 
-> TIP: If you want to check and validate a grammar rule you just wrote, you can use the ANTLR 4 plugin of your IDE. See Setting up your development environment section for more details.
+    - **BallerinaLexer.g4:** contains the lexer rules for Ballerina grammar. Lexer is responsible for tokenizing an input Ballerina source code.
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]:
```suggestion
    - **BallerinaLexer.g4:** contains the lexer rules for Ballerina grammar. The lexer is responsible for tokenizing an input Ballerina source code.
```

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -1,195 +1,119 @@
...
+    ```bash 
+    java -jar <PATH-TO-ANTLR-JAR>/antlr-4.5.3-complete.jar *.g4 -package org.wso2.ballerinalang.compiler.parser.antlr4 -o <BALLERINA_PROJECT_ROOT>/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/antlr4/
+    ```
 
-### Ballerina Compiler 
-https://medium.com/@sameera.jayasoma/ballerina-compiler-design-3406acc2476c
+    **Info:** The above command will autogenerate some Java classes. The Ballerina AST builder is written on top of the auto-generated `BallerinaParserBaseListener.java` class. Thus, if any new rules are added to the `BallerinaParser.g4`, above command will generate new methods in the `BallerinaParserBaseListener.java` and you need to override those newly-added methods inside the `BLangParserListener.java` accordingly.
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]:
```suggestion
    **Info:** The above command will autogenerate some Java classes. The Ballerina AST builder is written on top of the auto-generated `BallerinaParserBaseListener.java` class. Thus, if any new rules are added to the `BallerinaParser.g4`, the above command will generate new methods in the `BallerinaParserBaseListener.java` and you need to override those newly-added methods inside the `BLangParserListener.java` accordingly.
```

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -1,195 +1,119 @@
...
-Import the Ballerina project to IntelliJ IDEA similar to any other Gradle project. 
+- Join the conversations at:
 
-* Navigate to `File | Open` menu.
-* Browse the filesystem and Select Ballerina Project's root directory.
-   * If prompted to import Gradle project select `Auto import`.
-   * Set Project SDK as Java 1.8
+    - [StackOverflow](https://stackoverflow.com/questions/tagged/ballerina): to get help with Ballerina (use the Ballerina tag for any of your questions)
+    - [Slack](https://ballerina.io/community/slack/): for real-time discussions with the team and community
+    - [Ballerina-Dev Google Group](https://groups.google.com/forum/#!forum/ballerina-dev): (developer team mailing list) to discuss Ballerina roadmap, features, and related issues
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]: IMO, this is not just the "Developer team mail list". Even ballerina users can join and interact here. So it is more likely a community mailing list

**Comment by praneesha** [comment]
> @praneesha We don't need these icons here. Those were in website repo as it is generating a web page where presentation matters. So please have the details with links without icons and that will give the idea in simple format.

Changed them back to links.

File: CONTRIBUTING.md (hunk 8 above)
Comment (praneesha) [reply CONTRIBUTING.md]: Removed border as all HTML features are being removed when MD files are rendered to maintain simplicity.

File: CONTRIBUTING.md (hunk 10 above)
Comment (praneesha) [reply CONTRIBUTING.md]: Removed that item from the ToC as we do not have that section now.

File: CONTRIBUTING.md (hunk 11 above)
Comment (praneesha) [reply CONTRIBUTING.md]: Fixed as : - [Build the source code](#build-the-source-code)

File: CONTRIBUTING.md (hunk 12 above)
Comment (praneesha) [reply CONTRIBUTING.md]: Fixed as: - [Set up the ANTLR4 plugin](#set-up-the-antlr4-plugins)

File: CONTRIBUTING.md (hunk 17 above)
Comment (praneesha) [reply CONTRIBUTING.md]: Changed to: (community mailing list)

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -1,195 +1,115 @@
...
 
-##### Importing a Ballerina Project
+-  Start with GitHub issues that can be fixed easily:
+    - Browse issues labeled "good first issue" in the <a href="https://github.com/ballerina-platform/ballerina-lang/issues">ballerina-lang</a> repo.
+    - Use comments on the issue itself to indicate that you will be working on it and get guidance and help.
 
-All Ballerina repositories are developed as Maven or Gradle projects. So you can import any Ballerina project to Eclipse similar to any Maven or Gradle project. 
+## Set up the prerequisites
+1. Download [Ballerina](https://ballerina.io) and go through the [learning resources](https://ballerina.io/learn).
+2. Read the <a href="https://ballerina.io/code-of-conduct">Ballerina Code of Conduct</a>.
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]: Shouldn't we move these two points to "Get Started". Nothing to setup in the given two points. Those are just readings.

File: CONTRIBUTING.md [hunk 19]
```diff
@@ -1,195 +1,115 @@
...
-* Browse the filesystem and Select Ballerina Project's root directory.
-   * If prompted to import Gradle project select `Auto import`.
-   * Set Project SDK as Java 1.8
+- Join the conversations at:
 
-##### Useful IDEA Plugins
+    - [StackOverflow](https://stackoverflow.com/questions/tagged/ballerina): to get help with Ballerina (use the Ballerina tag for any of your questions)
+    - [Slack](https://ballerina.io/community/slack/): for real-time discussions with the team and community
+    - [Ballerina-Dev Google Group](https://groups.google.com/forum/#!forum/ballerina-dev): (community mailing list) to discuss Ballerina roadmap, features, and related issues
+    - [Twitter](https://twitter.com/ballerinalang): to tweet about Ballerina (use the “#ballerinalang” hashtag)
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]: Why GitHub is removed from here?  Github is not all about reporting bugs. Contributors can join on conversations on new features etc.

File: CONTRIBUTING.md [hunk 20]
```diff
@@ -1,195 +1,115 @@
...
+This plugin will be useful to check and validate a grammar rule you wrote. For instructions on installing the ANTLR4 plugin based on your preferred IDE/Editor (which supports ANTLR 4.x version), go to [ANTLR Documentation](https://www.antlr.org/tools.html).
 
-### Working With the Ballerina Grammar
+## Submit your contribution
 
-Ballerina grammar has been implemented using ANTLR4. It is recommended to have some basic understanding on ANTLR4 grammar syntax and concepts, before working with the Ballerina grammar. “Parr, Terence (January 15, 2013), The Definitive ANTLR 4 Reference” is a good reference to get started with.
+1. Do your changes in the source code.
+    
+    <details>
+    <summary>If you are contributing to Ballerina Grammar:</summary>
```
Comment (anupama-pathirage) [inline CONTRIBUTING.md]: For me this sentence doesn't make much sense on collapse mode.  Shall we reword it to `Contributing to Ballerina Grammer:`

File: CONTRIBUTING.md (hunk 18 above)
Comment (praneesha) [reply CONTRIBUTING.md]: Added the 2 points under the ""Get Started" section and removed the "Set up the prerequisites" section.

File: CONTRIBUTING.md (hunk 19 above)
Comment (praneesha) [reply CONTRIBUTING.md]: Added the below:
- [GitHub](https://github.com/ballerina-platform/ballerina-lang/issues): to file issues, join the conversations on new features, comment on other issues, and send your pull requests.

File: CONTRIBUTING.md (hunk 20 above)
Comment (praneesha) [reply CONTRIBUTING.md]: Done
